### ISSUE TITLE
Do not recommend `jet` grouping if there is no native counterpart

### ISSUE THREAD

**Issue by muvaf** [crossplane-contrib/terrajet#169]
### What problem are you facing?

We have released AWS, Azure and GCP providers built with Terrajet with `provider-jet-*` name and `*.jet.crossplane.io` group suffix in order to not conflict with native providers when they are installed in the same cluster.

However, we don't really need to do that for providers whose native counterpart does not exist. At the end of the day, the provider running in the cluster satisfies all Crossplane features, in fact, more than the native ones in some cases thanks to automation. Also, it's straightforward [to switch](https://github.com/crossplane-contrib/provider-jet-aws/blob/4bddb2e/internal/controller/iam/role/zz_controller.go#L45) to an implementation where external API is directly called instead of Terraform CLI, so we'll see providers who have a mix of controller variants. So, unless there is a counterpart that calls APIs directly that we'd like to differentiate, I don't think we should recommend a naming convention different than usual Crossplane provider naming conventions.

Overall, we shouldn't the [cost of confusion](https://github.com/crossplane/crossplane/blob/master/design/design-doc-provider-strategy.md#decision-for-initial-releases) if we don't have to.

### How could Terrajet help solve your problem?

We can update the guide to use usual naming conventions, for example `provider-<short name>` instead of `provider-jet-<short name>`. Also update the defaults [here](https://github.com/crossplane-contrib/terrajet/blob/8ddfb30/pkg/config/provider.go#L151).

**Comment by negz** [comment]
I'd prefer us to be consistent, and stick the the `jet` distinction even in cases where there is only a `jet` variant. My rationale:

* More consistent, is to me, less confusing. Jet providers always have jet in the name is conceptually simpler than "jet providers have jet in the name only when it doesn't clash with an existing provider".
* I'm (still) not bought into the idea that we'll have single provider binaries that are a mix of jet generated controllers and other implementations. We've discussed this at some length before, but to me it comes down to wanting a clear distinction along the lines of operational characteristics. Forking Terraform binaries for example presumably has debugging and performance implications that are different from making API calls via a native SDK.
* We think the long term future of providers does not involve wrapping Terraform. if `provider-foo` does wrap Terraform and we're not open to mixing and matching controller implementations inside a single package/binary then that leaves us stuck having to figure out what to call the not-Terraform variant of `provider-foo` when it eventually surfaces.

**Comment by turkenh** [comment]
I believe one of the key questions to make a decision here is how much we could see _terrajet as an implementation detail_. Because, if we could claim that, then having a single provider would be the best way to go simplifying to answer the following questions:

- As a platform developer, should I choose a jet or a native one while building my platform?
- As a platform developer, how can I migrate from jet to native or native to jet because of some reason.
- Would both be maintained or one would become uncared if the other option is good enough.
- As a cloud/saas provider, should I create/maintain a jet or native one?

So, I think the confusion that could be caused by having multiple flavors of the same provider is not less than the confusion caused by not being consistent with 3 big. 

With all the effort done in Terrajet to provide a consistent XRM compliant API so far, I think we are quite confident to say _Terraform is just an implementation detail_. We all are on the same page on native controllers being better than the ones calling Terraform behind the scenes due to multiple reasons but from an API POV, not much changes between them. 

In short, I believe we can provide the best user experience* by having a single one and could move to native in a per resource-based approach instead of having both options side by side.

_*_ 
- As a platform developer, I don't care how you're doing that, I just want this cloud resource with this desired state.
- As a platform developer, I don't want to deal with migration at some point.
- I know that this provider will be well maintained no matter it runs terraform or directly calls the API internally.
- As a cloud/saas provider, I can easily/immediately bootstrap a terrajet provider for my service. I can then move to native controllers resource by resource as needed.
